### PR TITLE
fix: update followSchema and followCardUiSchema

### PR DESCRIPTION
**Pull request by vivzhang**
affects: @esri/hub-common

1. Description:

Update followSchema, followCardUiSchema and related files to support follow card editor single mode item picker 

1. Instructions for testing:

1. Closes Issues: [8722](https://zentopia.esri.com/workspaces/sprint-board-60340b9721bc3a4ab6367db5/issues/gh/dc/hub/8722)

1. [x] Updated meaningful TSDoc to methods including Parameters and Returns, see [Documentation Guide](https://esri.github.io/hub-components/storybook/?path=/story/guides-documentation--page)

1. [x] used semantic commit messages
  
1. [x] PR title follows semantic commit format (**CRITICAL** if the title is not in a semantic format, the release automation will not run!)

1. [x] updated `peerDependencies` as needed. **CRITICAL** our automated release system can **not** be counted on to update `peerDependencies` so we _must_ do it manually in our PRs when needed. See the [updating peerDependencies](/RELEASE.md#Updating-peerDependencies) section of the release instructions for more details.

### CODE REVIEW

File: packages/common/src/core/schemas/internal/follow/FollowCardUiSchema.ts [hunk 1]
```diff
@@ -16,80 +39,100 @@ export const buildUiSchema = (
...
+              control: "hub-field-input-alignment",
+            },
+          },
+          {
+            labelKey: "followButton.buttonStyle",
+            scope: "/properties/buttonStyle",
+            type: "Control",
+            options: {
+              control: "hub-field-input-select",
+              labels: ["Solid Background", "Outline"],
```
Comment (juliannemarik) [inline packages/common/src/core/schemas/internal/follow/FollowCardUiSchema.ts:123]: Should these labels be translated? You should be able to use the same `{{...:translate}}` syntax as long as you use `interpolateTranslations` when you load the `uiSchema`.

File: packages/common/src/core/schemas/types.ts [hunk 2]
```diff
@@ -151,6 +151,7 @@ export enum UiSchemaMessageTypes {
 
 export interface IConfigurationSchema extends JSONSchema {
   type?: "object";
+  selectionMode?: "string";
```
Comment (juliannemarik) [inline packages/common/src/core/schemas/types.ts:154]: hmm, just curious why we're adding this to the `IConfigurationSchema` interface? I don't _think_ we want to do this?

File: packages/common/src/core/schemas/internal/follow/FollowCardUiSchema.ts [hunk 3]
```diff
@@ -1,12 +1,35 @@
...
+import { IUiSchema, UiSchemaRuleEffects } from "../../types";
+import {
+  WellKnownCatalog,
+  WellKnownCollection,
+  getWellKnownCatalog,
+} from "../../../../search/wellKnownCatalog";
+import { IUser } from "@esri/arcgis-rest-types";
+import { IHubCatalog } from "../../../../search/types/IHubCatalog";
+
+function getCatalogs(user: IUser): IHubCatalog[] {
```
Comment (juliannemarik) [inline packages/common/src/core/schemas/internal/follow/FollowCardUiSchema.ts:12]: minor: maybe add a quick doc comment for this function

**Review by juliannemarik — APPROVED** [review]
Left a few minor comments, but otherwise this is looking good - nice work!

File: packages/common/src/core/schemas/internal/follow/FollowCardUiSchema.ts (hunk 1 above)
Comment (vivzhang) [reply packages/common/src/core/schemas/internal/follow/FollowCardUiSchema.ts:123]: Good catch!

File: packages/common/src/core/schemas/types.ts (hunk 2 above)
Comment (vivzhang) [reply packages/common/src/core/schemas/types.ts:154]: Oops I left it here while I was testing the selection mode for the picker, no need to have it here anymore, removing...

**Pull request merged**